lib/matplotlib/contour.py

```diff
@@ -633,39 +608,36 @@ def _is_closed_polygon(X):
     return np.allclose(X[0], X[-1], rtol=1e-10, atol=1e-13)
 
 
-def _find_closest_point_on_path(lc, point):
+def _find_closest_point_on_path(xys, p):
     """
     Parameters
     ----------
-    lc : coordinates of vertices
-    point : coordinates of test point
+    xys : (N, 2) array-like
+        Coordinates of vertices.
+    p : (float, float)
+        Coordinates of point.
+
+    Returns
+    -------
+    d2min : float
+        Minimum square distance of *p* to *xys*.
+    proj : (float, float)
+        Projection of *p* onto *xys*.
+    imin : (int, int)
+        Consecutive indices of vertices of segment in *xys* where *proj* is.
```

Comment:
Let's clarify the edge-cases
~~~
Segments are thought to include their end-points; i.e if the closest point
on the path is a node in *xys* with index *i*, this returns ``(i-1, i)``.
For the special case that *xys* is a single point, this returns ``(0, 0)``.
~~~

Reply:
yup